src/ng/compile.js

```diff
@@ -1877,6 +1877,36 @@ function $CompileProvider($provide, $$sanitizeUriProvider) {
         return value || null;
       }
 
+      function setupControllers($element, attrs, transcludeFn, controllerDirectives, isolateScope, scope) {
```

Comment:
by adding `newIsolateScopeDirective` as a parameter, this can be moved out of the closure (or I am missing something?)